### DISCUSSION TITLE
From kg/ha to t/ha units conversion

### DISCUSSION THREAD

**Discussion by samaranin** [aquacropos/aquacrop#79] [General]
In `aquacrop\timestep\run_single_timestep.py` [in line 426](https://github.com/aquacropos/aquacrop/blob/master/aquacrop/timestep/run_single_timestep.py#L426) I've found code
```python
        # Calculate crop yield_ (tonne/ha)
        NewCond.yield_ = (NewCond.biomass / 100) * NewCond.harvest_index_adj
```
The documentation says that biomass value is in kg/ha units and yield_ value in t/ha.
Is 100 a correct value? Or there should be 1000? Or maybe `harvest_index_adj` makes the additional conversion in this case?
```python
        # Calculate crop yield_ (tonne/ha)
        NewCond.yield_ = (NewCond.biomass / 1000) * NewCond.harvest_index_adj
```
Have I missed something?
@pacs27 can you help with this, please?

**Comment by pacs27** [comment]
Hi @samaranin,

If you look at how biomass accumulation is calculated, the unit of it has to be g/m2 which is equivalent to 0.01 tn/ha.: https://github.com/aquacropos/aquacrop/blob/fbce3f355ca4e9de86da6b193fae402012f229ad/aquacrop/solution/biomass_accumulation.py#L104

In the matlab version of aquacrop (on which this repository is based), the crop water productivity is in g/m2, so the biomass has to have the same units. 

So the problem I think is that the biomass unit in the docs are not correct, but I don't know if I'm missing something. Maybe @thomasdkelly knows more about it.

**Reply by pacs27** [reply]
Watch this:
https://github.com/aquacropos/aquacrop/blob/6f6b93de6ae86d20fbc1a50ffd4c5cfde3fdcd8e/aquacrop/entities/crop.py#L446
https://github.com/aquacropos/aquacrop/blob/fbce3f355ca4e9de86da6b193fae402012f229ad/aquacrop/solution/biomass_accumulation.py#L91
https://github.com/aquacropos/aquacrop/blob/fbce3f355ca4e9de86da6b193fae402012f229ad/aquacrop/solution/biomass_accumulation.py#L104
https://github.com/aquacropos/aquacrop/blob/fbce3f355ca4e9de86da6b193fae402012f229ad/aquacrop/solution/biomass_accumulation.py#L109

**Reply by samaranin** [reply]
Thanks a lot for the answer. 
I got low biomass values in my experiments and looked for a reason. And when I found this formula and units in the documentation thought that might be my problem.
I'll look more precisely at the code you have mentioned and will run some more simulations. 
Thanks again for the explanation.

**Comment by Rachael001l** [comment]
Hi @samaranin ,
I also get a low biomass, I think the simulated biomass should be multiplied by 10 so that can close to the actual result.
But I don't know if it's right. Have you solved this problem?
If you get the same thought as me please tell me. Thank you so much.
